staging/src/k8s.io/kms/encryption/aesgcm.go

```diff
@@ -0,0 +1,100 @@
+package encryption
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"errors"
+)
+
+var (
+	ErrEmptyPlaintext = errors.New("plaintext for encryption is empty")
+)
+
+// AESGCM is a struct that contains a key of random bytes and additional
+// information to re-use it in a safe way.
+type AESGCM struct {
+	key []byte
+
+	aesgcm cipher.AEAD
+}
+
+// NewAESGCM returns a pointer to a AESGCM with a key and cipher.AEAD created.
+func NewAESGCM() (*AESGCM, error) {
+	key, err := randomBytes(keySize)
+	if err != nil {
+		return nil, err
+	}
+
+	return FromKey(key)
+}
+
+func FromKey(key []byte) (*AESGCM, error) {
```

Comment:
Could you add a comment to all the exported functions? (relevant for all the files in the PR)